workers/python/python/extraction/unstructured.py

```diff
@@ -0,0 +1,56 @@
+import asyncio
+
+from unstructured_ingest.v2.pipeline.pipeline import Pipeline
+from unstructured_ingest.v2.interfaces import ProcessorConfig
+from unstructured_ingest.v2.processes.partitioner import PartitionerConfig
+from unstructured_ingest.v2.processes.connectors.fsspec.s3 import (
+    S3IndexerConfig, S3DownloaderConfig, S3ConnectionConfig, S3AccessConfig)
+from unstructured_ingest.v2.processes.connectors.fsspec.s3 import (
+    S3ConnectionConfig, S3AccessConfig, S3UploaderConfig)
+from unstructured_ingest.v2.processes.chunker import ChunkerConfig
+
+from config import config
+from enums import ExtractionBackend
+from database import database
+
+EXTRACTION_DIR = "unstructured"
+
+S3_URL = f"s3://{config.s3_bucket_file_storage}"
+
+
+async def unstructuredExtraction(file):
+    storage_id = file["storageId"]
+
+    s3_connection_config = S3ConnectionConfig(
+        endpoint_url=config.s3_endpoint,
+        access_config=S3AccessConfig(
+            key=config.s3_access_key_id,
+            secret=config.s3_secret_access_key
+        )
+    )
+    pipeline = Pipeline.from_configs(
+        context=ProcessorConfig(),
+        indexer_config=S3IndexerConfig(remote_url=f"{S3_URL}/{storage_id}"),
+        downloader_config=S3DownloaderConfig(),
+        source_connection_config=s3_connection_config,
+        partitioner_config=PartitionerConfig(
+            partition_by_api=config.extraction_backend == ExtractionBackend.UNSTRUCTURED_API,
+            partition_endpoint=config.unstructured_api_url,
+            api_key=config.unstructured_api_key
+        ),
+        chunker_config=ChunkerConfig(
+            chunk_by_api=config.extraction_backend == ExtractionBackend.UNSTRUCTURED_API,
+            chunking_endpoint=config.unstructured_api_url,
+            chunk_api_key=config.unstructured_api_key,
+            chunking_strategy="by_title",
```

Comment:
We should make sure that the length of text doesn't exceed maximum tokens of the embedding model, is there such strategy or config option?

Reply:
Default is max of 500 characters and 0 overlap which is roughly what we want. We currently don't pre-chunk with setting for a specific model, I don't expect we will have to either.